x-pack/solutions/observability/plugins/observability/server/lib/rules/custom_threshold/custom_threshold_executor.ts

```diff
@@ -298,6 +302,22 @@ export const createCustomThresholdExecutor = ({
     alertsClient.setAlertLimitReached(hasReachedLimit);
     const recoveredAlerts = alertsClient.getRecoveredAlerts() ?? [];
 
+    let groupingObjectForRecovered: Record<string, object> = {};
+
+    // extracing group by fields from kibana.alert.rule.params,
+    // since all recovered alert documents will have same group by fields,
+    // we are only checking first recovered alert document
+    if (recoveredAlerts.length > 0) {
+      const alertHit = recoveredAlerts[0].hit;
+      const ruleParamsOfRecoveredAlert = alertHit?.[ALERT_RULE_PARAMETERS] as EvaluatedRuleParams;
+      const groupByFields = ruleParamsOfRecoveredAlert?.groupBy;
```

Comment:
Nit: maybe we can add the above comment for this line, at the start, I thought the whole if block is only related to one recovered alert.

Reply:
It is kind of for one recovered alert. The purpose is to get group by fields and since it will be same for all recovered alerts, we can check any one of them.